### PR TITLE
Don't include message if empty

### PR DESCRIPTION
**Pull request by gjoseph92**
Empty messages show up like this in Stackdriver logging (at least for me):

Screen Shot 2020-04-07 at 10 50 52 PM

Nicer to just skip them entirely.

### CODE REVIEW

**Comment by balboah** [comment]
Hi, thanks for the pull request. In what circumstances do you get empty messages? I'm not convinced that this filtering should happen in here but perhaps rather in the application itself?
Consider for example that you debug log something that you wouldn't expect to be empty, if you don't get anything you'd perhaps think there's something wrong with the log setup itself?

**Comment by gjoseph92** [comment]
Fair point.

> In what circumstances do you get empty messages?

Mostly just like the case I'm testing: `logger.WithField("foo", "bar").Info()`, where you don't pass any message into the log method itself, but you're just logging for the fields.

In my application, we do this sometimes when there's no relevant message for the event; something like:

```go
logger.withFields(map[string]interface{} {
  "event": "serve_request",
  "elapsed": timeElapsed,
  "user": user_id,
  "object": object_id,
}).Info()
```
In my application, we're also handling structured JSON logs from a different system running in a subprocess. Sometimes that system produces log entries like the one above, where there's no "message" field, and no definitive mapping from some other named field to the message.

> I'm not convinced that this filtering should happen in here but perhaps rather in the application itself?

I'd be happy to do this, but I made this PR because it's not possible. Even if I don't supply a message, an empty one is still added. I'd have to take the output of this formatter, decode it from JSON again, delete the message field if it's empty, and re-encode it to JSON. At that point, it kind of defeats the purpose of using the library.

Basically, my opinion is that if you don't pass in a message, you just shouldn't get a message—the field shouldn't be implicitly added for you.

> Consider for example that you debug log something that you wouldn't expect to be empty, if you don't get anything you'd perhaps think there's something wrong with the log setup itself?

That's a valid concern. I'd say that it is possible to work around that case:
```go
if message == "" {
  message = "<nil>"
}
```
whereas there's no workaround for my case of wanting to drop the message.

**Comment by balboah** [comment]
Thanks for the explanation 👍
